routers/logout_filter.go

```diff
@@ -0,0 +1,41 @@
+package routers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/beego/beego/context"
+	"github.com/beego/beego/logs"
+	"github.com/casdoor/casdoor/conf"
+)
+
+var (
+	logoutMinutes   = time.Minute * 30
+	cookie2LastTime = make(map[string]time.Time, 1)
+)
+
+func init() {
+	logoutMinutes_int, err := conf.GetConfigInt64("logoutMinutes")
+	if err != nil {
+		logs.Info(fmt.Sprintf("get logoutMinutes failed, err:%v. use default time duration: 30 minutes", err))
+	} else {
+		logoutMinutes = time.Minute * time.Duration(logoutMinutes_int)
+	}
+}
+
+func LogoutFilter(ctx *context.Context) {
+	owner, name := getSubject(ctx)
+	if owner != "anonymous" && name != "anonymous" {
+		sessionId := ctx.Input.CruSession.SessionID()
+		currentTime := time.Now()
+		if cookieTime, exist := cookie2LastTime[sessionId]; exist && cookieTime.Add(logoutMinutes).Before(currentTime) {
+			delete(cookie2LastTime, sessionId)
```

Comment:
map is not thread safety, maybe we should add lock or use sync.Map to solve it.

Reply:
or we can add last time to session to avoid use map

Reply:
@dacongda I use sync.map to ensure thread safety.
Because all requests need to determine whether they have not been operated for a long time, locking the entire dictionary during the judgment may cause performance bottlenecks. However, beego session data has the problem of update overwriting in concurrent scenarios. Therefore, I did not take these two methods.